openshift/oc-mirror
# CFE-768: Update to allow for incremental and pruning for oci feature

# Description

This change allows for the the incremental and pruning functionality to work for the oci feature

## Type of change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The update was tested as follows 

- Add an oci catalog in the imagesetconfig (i.e cincinnati-operator) and mirror to remote registry
- Update oci catalog in the imagesetconfig (remove cincinbati-operator and add openshift-logging) and mirror
- Verfiy the mapping.txt had correct entry
- Verify the .metatadat.json file sequence was bumped
- Verify the console logs for both oc-mirror and remote registry (see below)

```
Pruning 6 manifest(s) from repository test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Pruning 3 manifest(s) from repository test-lmz/openshift-update-service/cincinnati-operator-bundle
Pruning 6 manifest(s) from repository test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:1d3bde7bc3ce4f6acb79307e67aa0342b91fca5ff69bf38e2b0a1884c5cd7db6 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:2743f74cc102c059d7c1dc6698eb947cca75a3cc4230905f09703666ae7a4dbf from repo test-lmz/openshift-update-service/cincinnati-operator-bundle
Deleting manifest sha256:21e32b201bba902cc730dc43943d9cb6c4bab40afc5a148613ae51176c21cd85 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:50566d6670315bc3a4526644404c8e8ab5bd1e1709c1fd9fcde5a37abcb7cb04 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:389c339253271477440c5f470c08a3b149f1795647b1df7b104ae95fc63af22c from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:3feccb95e912f2d710958e701a1b52acf16a676fde9a8219491ad9381e6284ea from repo test-lmz/openshift-update-service/cincinnati-operator-bundle
Deleting manifest sha256:66db743f2d61dddab7363ab36252812ffb63653510124c3b32d26f0ddfeb5def from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:4bbed256e300088500ae3eb8475b3457d7a6765999e1102dc482e086e2cc0b43 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:71b8f7816a01503a4af5d6f93b9b3a057f6c011aef3b560b408ee4d7abf02165 from repo test-lmz/openshift-update-service/cincinnati-operator-bundle
Deleting manifest sha256:c8f156f73bb3e5fd92c024513c9446ed30d5231e33ef0d0952b365a1fbd69db4 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:6e65da919ed05274e4a0101f67a149188801b715cf073dd517d0dd311a4572a4 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:ce77a27ae4b7b219e44111b6322a3387f75c23fa39ecd94017d2d4bc68bf4a5c from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:a2cc2f9d3acd147ced68d8657cf8dd6189be242b4e82852ea8ef002d5f55b31e from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Deleting manifest sha256:ce7b16803bb2bd058014a3783ed1f0a8a93819dd5769fcd8772228cf96814bf9 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8-operator
Deleting manifest sha256:aea49b2145d59912757f2e371db2e23a9ba20cfa54d0c7132931e16d35655e40 from repo test-lmz/openshift-update-service/openshift-update-service-rhel8
Rendering catalog image "localhost:5000/lmz-test/home/lzuccarelli/go/src/github.com/openshift/oc-mirror/rhopi-412:2cde02" with file-based catalog 
```

Registry log (top entry)

```
[16/Feb/2023:15:10:57 +0100] "DELETE /v2/test-lmz/openshift-update-service/openshift-update-service-rhel8-operator/manifests/sha256:ce7b16803bb2bd058014a3783ed1f0a8a93819dd5769fcd8772228cf96814bf9 HTTP/1.1" 202 0 "" "go-containerregistry/v0.10.0"
INFO[13249] response completed                            go.version=go1.19.5 http.request.host="localhost:5000" http.request.id=3d7cf0cd-7f03-438e-a9fc-0fb7431add9a http.request.method=DELETE http.request.remoteaddr="[::1]:33704" http.request.uri="/v2/test-lmz/openshift-update-service/openshift-update-service-rhel8/manifests/sha256:aea49b2145d59912757f2e371db2e23a9ba20cfa54d0c7132931e16d35655e40" http.request.useragent=go-containerregistry/v0.10.0 http.response.duration="259.493µs" http.response.status=202 http.response.written=0 instance.id=18c670a9-9bee-4d1b-9f24-2765bebb6c74 service=registry version=v3.0.0+unknown
::1 - - [16/Feb/2023:15:10:57 +0100] "DELETE /v2/test-lmz/openshift-update-service/openshift-update-service-rhel8/manifests/sha256:aea49b2145d59912757f2e371db2e23a9ba20cfa54d0c7132931e16d35655e40 HTTP/1.1" 202 0 "" "go-containerregistry/v0.10.0"
```



**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:
